src/consensus/aft/raft.h

```diff
@@ -1796,6 +1789,13 @@ namespace aft
       // and has a genesis or recovery tx as the last transaction
       election_index = last_committable_index();
 
+      // A newly elected leader must not advance the commit index until a
+      // transaction in the new term commits. We achieve this by clearing our
+      // list committable indices - so nothing from a previous term is now
+      // considered committable. Instead this new primary will shortly produce
+      // their own signature, which _will_ be considered committable.
+      committable_indices.clear();
```

Comment:
This is the fix, to prevent previous entries being considered committable when this node becomes primary.

Reply:
This change not only has the desired effect that a new primary cannot advance its commit index until a transaction in the new term commits, but also that the leader may truncate its log if an update term message is received before the leader produces its own signature (see https://github.com/microsoft/CCF/issues/5674#issuecomment-1769868427).